File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        long ans = 0;
        int dummy = Math.abs(x);

        while(dummy > 0){
            ans = (ans * 10) + dummy % 10;
            dummy /= 10;
            System.out.println(ans);
        }

        if(x < 0)
            ans *= (-1);

        if(ans > Math.pow(2, 31) || ans < -Math.pow(2,31))
            return 0;

        return (int)ans;
    }
}

File: 7-reverse-integer/reverse-integer.cs
public class Solution {
    public int Reverse(int x) {
        bool isNegative = x < 0;

        if(isNegative)
            x *= (-1);
        
        long ans = 0;

        while(x > 0){
            int rem = x % 10;
            x /= 10;
            ans = (ans * 10) + rem;
        }

        if(isNegative)
            ans *= (-1);
        
        if(ans > int.MaxValue || ans < int.MinValue)
            return 0;
        
        return (int)ans;
    }
}